worldedit-core/src/main/java/com/sk89q/worldedit/session/request/RequestExtent.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * WorldEdit, a Minecraft world manipulation toolkit
+ * Copyright (C) sk89q <http://www.sk89q.com>
+ * Copyright (C) WorldEdit team and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.sk89q.worldedit.session.request;
+
+import com.sk89q.worldedit.EditSession;
+import com.sk89q.worldedit.WorldEditException;
+import com.sk89q.worldedit.entity.BaseEntity;
+import com.sk89q.worldedit.entity.Entity;
+import com.sk89q.worldedit.extent.Extent;
+import com.sk89q.worldedit.function.operation.Operation;
+import com.sk89q.worldedit.math.BlockVector2;
+import com.sk89q.worldedit.math.BlockVector3;
+import com.sk89q.worldedit.regions.Region;
+import com.sk89q.worldedit.util.Location;
+import com.sk89q.worldedit.world.biome.BiomeType;
+import com.sk89q.worldedit.world.block.BaseBlock;
+import com.sk89q.worldedit.world.block.BlockState;
+import com.sk89q.worldedit.world.block.BlockStateHolder;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+public class RequestExtent implements Extent {
+
+    private EditSession extent;
+
+    protected Extent getExtent() {
+        if (extent == null) {
```

Comment:
Since you're caching it now, are you sure the EditSession's lifetime is guaranteed to be greater than that of the RequestExtent?
What if the player leaves?

The problem with lazy creation is that problems with it will only show up much much later.

Reply:
Oh I just saw you null that field again in `.commit()`. I don't know WE's new structure very well. Is that the function that is  called at the end of a time slice or something?

Reply:
Yea, this doesn't actually work yet. You miss out of the discord discussion, but essentially this extent tends to be used for masks which mainly do gets, not sets, so this extent isn't actually committed anywhere. The "correct" place to check would be when `Request#setEditSession` and `Requests#reset` are called.